pkg/messageprocessors/javascript/javascript_test.go

```diff
@@ -539,8 +543,23 @@ func TestDecodeUplink(t *testing.T) {
 			}
 			`
 		err := host.DecodeUplink(ctx, ids, nil, message, script)
-		a.So(err, should.NotBeNil)
-		a.So(errors.IsInvalidArgument(err), should.BeTrue)
+		a.So(err, should.BeNil)
+
+		a.So(message.NormalizedPayload, should.HaveLength, 1)
+		parsedMeasurements, err := normalizedpayload.Parse(message.NormalizedPayload)
+		a.So(err, should.BeNil)
+		measurements := make([]normalizedpayload.Measurement, len(parsedMeasurements))
+		for i, m := range parsedMeasurements {
+			measurements[i] = m.Measurement
+		}
+		a.So(measurements, should.Resemble, []normalizedpayload.Measurement{
+			{
+				Air: &normalizedpayload.Air{},
```

Comment:
I think the measurement should be skipped completely, otherwise you're providing the subscribers with a measurement that would not pass validation (think of a measurement which has an invalid zero value).

Reply:
Yeah actually I just deleted the commit that omitted empty objects cause I think it's not a good idea to do that as part of validation.

All values are optional anyway, so default values will not cause invalid values.

The issue is that if a developer returns:

```js
return {
  data: {
    air: {}
  }
}
```

We would keep `air`, but if there was an invalid temperature in `air` (and no other valid field), we would remove `air`. Hence I deleted that commit because that is ambiguous.

I'm not against omitting empty objects, but if we do it, it would come after validation. I just don't know what the value is, except that it looks cleaner. Would there be objects in the future which presence already indicate something? Like `location`; the device _tried_ getting a GPS fix but didn't get any, vs not having `location` at all?

Reply:
For location, historically payload formatters had a `fix` or `hasFix` field that would be false when an attempt was made, but was unsuccessful. I don't think that we should give 'presence' semantics to these empty measurements - I don't consider the empty measurement to be an error, but I do think it should be stripped.

Reply:
I agree with @adriansmares, emitting empty fields will be useless for integrations. Imagine that an application developer stores the normalized payloads in a database as they arrive, without any checking or format modification, that database will be wasting storage on data without any meaningful value.

Reply:
Okay, fine with me as well!